OpenRA.Mods.Common/Traits/BlocksProjectiles.cs

```diff
@@ -58,7 +58,12 @@ public static bool AnyBlockingActorsBetween(World world, WPos start, WPos end, W
 				if (!blockers.Any())
 					continue;
 
-				var hitPos = WorldExtensions.MinimumPointLineProjection(start, end, a.CenterPosition);
+				// FindActorsOnLine only finds actors with health trait, so this actor is guaranteed to have it
+				var healthInfo = a.Info.TraitInfo<HealthInfo>();
+				var vector = (a.CenterPosition - source) * healthInfo.Shape.InnerRadius.Length / (a.CenterPosition - source).Length;
```

Comment:
The overall goal is to find the "first" intersection of the line defined by (end - start) with the hitbox shape (do you really want the edge? or do you want the middle?).  You can't do this properly because the `IHitShape` interface doesn't expose the geometry info you need – `InnerRadius` is still a rather crude workaround for this that i'm not really happy about.

A better way to do this will be to define a `bool IHitShape.TryGetIntersection(WPos start, WPos end, out WPos intersection)` so that each shape can do its own logic to find the appropriate intersection point (e.g. circle vs line test for circles, line vs line tests for rectangle).